zerver/tests/test_message_summary.py

```diff
@@ -0,0 +1,103 @@
+from zerver.views.message_summary import INPUT_COST_PER_GIGATOKEN, OUTPUT_COST_PER_GIGATOKEN
+
+warnings.filterwarnings("ignore", category=UserWarning, module="pydantic")
+warnings.filterwarnings("ignore", category=DeprecationWarning, module="pydantic")
+warnings.filterwarnings("ignore", category=DeprecationWarning, module="litellm")
+# Avoid network query to fetch the model cost map.
+os.environ["LITELLM_LOCAL_MODEL_COST_MAP"] = "True"
+import litellm
+
+# Fixture file to store recorded responses
+LLM_FIXTURES_FILE = "zerver/tests/fixtures/litellm/summary.json"
+
+
+class MessagesSummaryTestCase(ZulipTestCase):
+    @override
+    def setUp(self) -> None:
+        super().setUp()
+        self.user = self.example_user("iago")
+        self.topic_name = "New feature launch"
+        self.channel_name = "Zulip features"
+
+        self.login_user(self.user)
+        self.subscribe(self.user, self.channel_name)
+        content = "Zulip just launched a feature to generate summary of messages."
+        self.send_stream_message(
+            self.user, self.channel_name, content=content, topic_name=self.topic_name
+        )
+
+        content = "Sounds awesome! This will greatly help me when catching up."
+        self.send_stream_message(
+            self.user, self.channel_name, content=content, topic_name=self.topic_name
+        )
+
+        if settings.GENERATE_LITELLM_FIXTURES:  # nocoverage
+            self.patcher = mock.patch("litellm.completion", wraps=litellm.completion)
+            self.mocked_completion = self.patcher.start()
+
+    @override
+    def tearDown(self) -> None:
+        if settings.GENERATE_LITELLM_FIXTURES:  # nocoverage
+            self.patcher.stop()
+        super().tearDown()
+
+    def test_summarize_messages_in_topic(self) -> None:
+        narrow = orjson.dumps([["channel", self.channel_name], ["topic", self.topic_name]]).decode()
+
+        if settings.GENERATE_LITELLM_FIXTURES:  # nocoverage
+            # NOTE: You need have proper credentials in zproject/dev-secrets.conf
+            # to generate the fixtures. (Tested using aws bedrock.)
+            # Trigger the API call to extract the arguments.
+            self.client_get("/json/messages/summary", dict(narrow=narrow))
+            call_args = self.mocked_completion.call_args
+
+            # Once we have the arguments, call the original method and save its response.
+            response = self.mocked_completion(**call_args.kwargs).json()
+            with open(LLM_FIXTURES_FILE, "wb") as f:
+                fixture_data = {
+                    # Only store model and messages.
+                    # We don't want to store any secrets.
+                    "model": call_args.kwargs["model"],
+                    "messages": call_args.kwargs["messages"],
+                    "response": response,
+                }
+                f.write(orjson.dumps(fixture_data, option=orjson.OPT_INDENT_2) + b"\n")
+        else:
```

Comment:
Probably the early return pattern would be better here.

Reply:
I did that but then preserved the indentation by changing the `settings` edits to use `with self.settings`, which is important because it correctly cleans up the patching.